### PR TITLE
Diff doc

### LINKED ISSUE #370: add Reference to diff
adding reference material to diff.md

### PR DESCRIPTION
**Pull request by jrbr**
# Advanced HW 5 submission

Closes #370

Name: Joe Brenner
Uniqname: jrbr

### CODE REVIEW

File: _commands/tools/diff.md [hunk 1]
```diff
@@ -37,5 +37,12 @@ $
 This outputs both files in 2 columns with a '|' marking lines with differences, similar to the standard output of 'sdiff'.
```
Comment (cyanliu) [inline _commands/tools/diff.md]: The pipe disappears D: It'll need to be escaped with a `\`, so the text in the markdown file should look something more like `... with a '\|' marking ...`

File: _commands/tools/diff.md [hunk 2]
```diff
@@ -2,23 +2,47 @@
...
 
-#### Example command
+#### `diff -y`
+
+This outputs both files in 2 columns with a '|' marking lines with differences, similar to the standard output of 'sdiff'.
+
+~~~ bash
+$ printf "working with github\nlets you share your work\ncollaboration" > first.txt
+$ printf "working with github\nlets you backup your data\nrefrigerator" > second.txt
+$ diff -y haiku1.txt haiku2.txt
```
Comment (cyanliu) [inline _commands/tools/diff.md]: oh and the two printf's redirect into files called "first.txt" and "second.txt" yet the diff command uses "haiku1.txt" and "haiku2.txt"

File: _commands/tools/diff.md [hunk 3]
```diff
@@ -2,23 +2,47 @@
...
 ~~~ bash
-*short* example of command usage and output
+$ printf "this\ndifference\n" > first.txt
+$ printf "this\nsimilarity\n" > second.txt
+$ diff first.txt second.txt
+2c2
+< difference
+---
+> similarity
+$
```
[the remark this answers is not in the document]
Comment (jrbr) [reply _commands/tools/diff.md]: which line are you referring to?

[the remark this answers is not in the document]
Comment (cameron-gagnon) [reply _commands/tools/diff.md]: Line 17. The one with just the `$` on it.

Also, comments on here appear just under the line that they were made on (for future reference. It's kinda confusing without knowing this).

File: _commands/tools/diff.md (hunk 1 above)
Comment (jrbr) [reply _commands/tools/diff.md]: all your comments were resolved. Thanks for the feedback!

File: _commands/tools/diff.md [hunk 4]
```diff
@@ -2,23 +2,47 @@
...
+
+~~~ bash
+$ printf "working with github\nlets you share your work\ncollaboration" > first.txt
+$ printf "working with github\nlets you backup your data\nrefrigerator" > second.txt
+$ diff -y haiku1.txt haiku2.txt
+working with github			working with github
+lets you share your work	      | lets you backup your data
+collaboration			      | refrigerator
+$
+~~~
```
[the remark this answers is not in the document]
Comment (jrbr) [reply _commands/tools/diff.md]: added

File: _commands/tools/diff.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (jrbr) [reply _commands/tools/diff.md]: Thanks! That's really good to know and isn't very clear initially

**Pull request merged**